src/main/java/org/hmhb/audit/DefaultAuditHelper.java

```diff
@@ -1,22 +1,38 @@
 package org.hmhb.audit;
 
+import javax.annotation.Nonnull;
+import javax.servlet.http.HttpServletRequest;
+
 import java.util.Date;
 
 import com.codahale.metrics.annotation.Timed;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import static java.util.Objects.requireNonNull;
+
 @Service
 public class DefaultAuditHelper implements AuditHelper {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultAuditHelper.class);
 
+    private final HttpServletRequest request;
```

Comment:
I was a little concerned that Spring would do a bad thing due to it using singletons (and associate the first IP with all later requests), however, I tested it out on several devices and it found the correct IP despite using the same instance of DefaultAuditHelper.

I suspect Spring is injecting a proxy to the real underlying request and probably doing something with ThreadLocals to associate things back.